### PR TITLE
Bitfields are loaded with the correct value when using `ActiveRecord.find`

### PR DESCRIPTION
**Pull request by shirish-pampoorickal**
Ensure that models with bitfields set to true initialize with correct values
- Due to a quirk in how ActiveRecord initializes models when loading them from the database, the bitfields are set to `false` by default.
  - When new-ing up records from the db, ActiveRecord bypasses the standard initializer and instead uses an internal `instantiate` method 
    - https://github.com/rails/rails/blob/b9ca94caea2ca6a6cc09abaffaad67b447134079/activerecord/lib/active_record/persistence.rb#L58-L72
- This doesn't affect the results of the getters and setters, since we manually define their behavior to inspect the bitfield column directly.
- What this does do, however, is "confuse" the underlying mutation tracker that keeps record of the models dirty attributes.
  - As such, when we set `foo = false` and check `.changes`, the bit column will show as changed, but `foo` will be missing.
  - If we then set `foo = true`, the bit column will be removed from `changes`, but `foo` now exists.
  - As such, when we set `foo = false` and check `.changes`, the bit column will show as changed, but `foo` will be missing.
  - If we then set `foo = true`, the bit column will be removed from `.changes`, but now `foo: [false, true]`.

Prefer using write_attribute over `send`ing
- Backfill specs for each lifecycle state of a model.

Update each bitfield set to define only a single after_find
- Creating this within the `added_instance_methods` block where the rest of the bit definitions happen would cause `after_find` to be defined for each individual bit.
- This change just avoids not only bloating the model, but eliminating unnecessary context switching for each method invocation.

Only try to load attributes if the bitfield column is included in the set of attributes
- When loading a model from the DB, we may not have loaded the bitfield column.
  - e.g. Loading a user model via `User.select(:id).first` would initialize a user without the `:bits` column.
  - As such, when our `after_find` runs, we would raise an error while attempting to load the value for `:bits`.
  - To resolve this, we only attempt to load the attribute if the model itself has the that attribute defined.

Prefer using internal method for grabbing value
- This was done since we may potentially have application code that may either overwrite the getter and produce unexpected behavior.
  - One such example would be mocking the getter for the bitfield which could cause issues for number of call expectations.

Refactor after_find logic to avoid unnecessary method calls
- Prefer using internal method to grab list of all current bitfield values.
- Only call `clear_attribute_changes` once for entire set of bitfields.

### CODE REVIEW

**Comment by a-bates** [comment]
Still to do on this is to make the `after_find` a single definition that loops over all bitfields in a single go, instead of running individual `after_find`s for each bit

File: gemfiles/activerecord_5.1.gemfile.lock
Comment (naganowl) [inline gemfiles/activerecord_5.1.gemfile.lock:61]: Was `bundler` supposed to be updated here?

**Review by naganowl — COMMENTED** [review]
Looks good, though I'd appreciate another set of eyes on this. Had just a minor question on this.

File: gemfiles/activerecord_5.1.gemfile.lock
Comment (a-bates) [reply gemfiles/activerecord_5.1.gemfile.lock:61]: We don't think it really matters, but we can bump it down if the upstream maintainer wants to it be locked to `2.0.1`

**Review by brandonduff — APPROVED** [review]
Thanks for fleshing out the test cases!

**Pull request merged**
